### PR TITLE
Correlation function doc fix

### LINKED ISSUE #740: Correlation function docs are equivocal
corr

The above should explain that this is not specific to galaxies or shear, but general for any spin-0 or spin-2 fields respectively.

### PR DESCRIPTION
**Pull request by damonge**
Closes #740

### CODE REVIEW

File: pyccl/correlation.py [hunk 1]
```diff
@@ -37,9 +37,11 @@ def correlation(cosmo, ell, C_ell, theta, corr_type='gg', method='fftlog'):
                                      calculate the angular correlation
                                      function (in degrees).
         corr_type (string): Type of correlation function. Choices:
-                            'GG' (galaxy-galaxy), 'GL' (galaxy-shear),
-                            'L+' (shear-shear, xi+),
-                            'L-' (shear-shear, xi-).
+                            'GG' (0x0), 'GL' (0x2),
+                            'L+' (2x2, xi+),
+                            'L-' (2x2, xi-), where the numbers refer to the
```
Comment (slosar) [inline pyccl/correlation.py]:
```suggestion
                            'L-' (2x2, xi-), where numbers refer to the
```
Do you need an article here? [asks clueless slavic speaker]

Comment (damonge) [reply pyccl/correlation.py]: Wow, didn't know you could commit a suggestion!
Yeah, sounds good to me [says clueless Espanish speaker].

File: pyccl/correlation.py [hunk 2]
```diff
@@ -54,12 +65,25 @@ def correlation(cosmo, ell, C_ell, theta, corr_type='gg', method='fftlog'):
...
+        if corr_type == 'gg':
+            type = 'NN'
+        elif corr_type == 'gl':
+            type = 'NG'
+        elif corr_type == 'l+':
+            type = 'GG+'
+        elif corr_type == 'l-':
+            type = 'GG-'
+        else:
+            raise ValueError("Unknown corr_type" + corr_type)
```
Comment (rmjarvis) [inline pyccl/correlation.py]: Probably want to add a space here.

**Comment by damonge** [comment]
Great, thanks!

**Pull request merged**